### ISSUE TITLE
PHP 8.1 support incomplete

### ISSUE THREAD

**Issue by SharkMachine** [squizlabs/PHP_CodeSniffer#3799]
**Code sample**
```php
<?php

/**
 * @param One&Two $param
 * @return void
 */
function sigh(One&Two $param): void
{
    // Do you really support PHP 8.1
}
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
```
 4 | ERROR | [x] Tag value for @param tag indented incorrectly; expected 2 spaces but found 1
   |       |     (Generic.Commenting.DocComment.TagValueIndent)

```

**Expected behavior**

No error

**Versions (please complete the following information):**
 - OS: Ubuntu 22.04
 - PHP: 8.1
 - PHPCS: 3.7.2
 - Standard: You can see it from above

**Comment by SharkMachine** [comment]
#3798

**Comment by fredden** [comment]
If you add a blank\* line between lines 4 & 5 in your sample, do you still get an error? I expect not.
Do you get the same error if the type is not a union type? I expect so.

This seems like an alignment issue, not a (lack of) PHP version support.

\* Not an completely blank line, as this is in a docblock, so you will need the correct indentation and an asterisk, but nothing else.

**Comment by SharkMachine** [comment]
> If you add a blank* line between lines 4 & 5 in your sample, do you still get an error? I expect not. Do you get the same error if the type is not a union type? I expect so.
> 
> This seems like an alignment issue, not a (lack of) PHP version support.
> 
> * Not an completely blank line, as this is in a docblock, so you will need the correct indentation and an asterisk, but nothing else.

PHPCS doesn't understand the required spacing when using intersection types.

**Comment by SharkMachine** [comment]
@fredden Instead of looking at my problem and help me to solve it, you wanted to belittle me.

**Comment by mrVrAlex** [comment]
Another case (+ PHP 8.2)

```
<?php

declare(strict_types=1);

namespace Test;

/**
 * Bla bla bla
 */
readonly class Example
{
    public function test() : void
    {
        echo '123';
    }
}
```

Show 
```
FOUND 2 ERRORS AFFECTING 2 LINES
------------------------------------------------------------------------------------------------------------------------------------
 7 | ERROR | [ ] The file-level docblock must follow the opening PHP tag in the file header
   |       |     (PSR12.Files.FileHeader.IncorrectOrder)
 9 | ERROR | [x] Header blocks must be separated by a single blank line (PSR12.Files.FileHeader.SpacingAfterBlock)
------------------------------------------------------------------------------------------------------------------------------------
```

If remove readonly OR put final - all will work.

**Comment by fredden** [comment]
@mrVrAlex that sounds like https://github.com/squizlabs/PHP_CodeSniffer/issues/3727 and not what @SharkMachine was trying to highlight. Can you confirm?

**Comment by jrfnl** [comment]
@mrVrAlex I agree with @fredden, your issue is unrelated to the one being discussed here and belongs with issue #3727.
Either way, I have pulled a fix now in https://github.com/squizlabs/PHP_CodeSniffer/pull/3816

**Comment by vladimir-borduja** [comment]
@jrfnl Apologies, that bother you. But could you please tell us when is the next release that will include these modifications? 

PS: PHP8.1 exited in November 2021, my team has been using it for more than 1 year, and we are still avoiding using read-only classes because our pipelines run code sniffer

**Comment by Fahani** [comment]
@jrfnl we are facing the same issue as @vladimir-borduja, any news about the next release?

Thank you!

**Comment by jrfnl** [comment]
> @jrfnl we are facing the same issue as @vladimir-borduja, any news about the next release?

@vladimir-borduja @Fahani Unfortunately no, we all (including me), will just have to wait until @gsherwood has some availability again... Also see #3861

**Comment by Tahiaji** [comment]
> Another case (+ PHP 8.2)
> 
> ```
> <?php
> 
> declare(strict_types=1);
> 
> namespace Test;
> 
> /**
>  * Bla bla bla
>  */
> readonly class Example
> {
>     public function test() : void
>     {
>         echo '123';
>     }
> }
> ```
> 
> Show
> 
> ```
> FOUND 2 ERRORS AFFECTING 2 LINES
> ------------------------------------------------------------------------------------------------------------------------------------
>  7 | ERROR | [ ] The file-level docblock must follow the opening PHP tag in the file header
>    |       |     (PSR12.Files.FileHeader.IncorrectOrder)
>  9 | ERROR | [x] Header blocks must be separated by a single blank line (PSR12.Files.FileHeader.SpacingAfterBlock)
> ------------------------------------------------------------------------------------------------------------------------------------
> ```
> 
> If remove readonly OR put final - all will work.

Another slightly weird way to fix this is to add a docblock between `<?php` and `declare(strict_types=1);`

**Comment by jrfnl** [comment]
@Tahiaji That case has already been fixed in `master` via #3816

**Comment by jrfnl** [comment]
FYI: A fix for [the issue reported by @mrVrAlex](https://github.com/squizlabs/PHP_CodeSniffer/issues/3799#issuecomment-1532870980) (PSR12/FileHeader sniff) is included in today's [PHP_CodeSniffer 3.8.0 release](https://github.com/PHPCSStandards/PHP_CodeSniffer/releases/tag/3.8.0).

As per #3932, development on PHP_CodeSniffer will continue in the [PHPCSStandards/PHP_CodeSniffer](https://github.com/PHPCSStandards/PHP_CodeSniffer) repository. If you want to stay informed, you may want to start "watching" that repo (or watching releases from that repo).

**Comment by SharkMachine** [comment]
This still being open doesn't look good for https://wiki.php.net/rfc/dnf_types :<
